src/zkevm_specs/ecc_circuit.py

```diff
@@ -0,0 +1,261 @@
+from __future__ import annotations
+from typing import List, NamedTuple, Tuple
+from py_ecc.bn128.bn128_curve import is_inf, is_on_curve, b
+
+from zkevm_specs.util.arithmetic import FP
+from .evm_circuit import EccTableRow
+from .util import ConstraintSystem, FQ, Word, ECCVerifyChip
+from zkevm_specs.evm_circuit.table import EccOpTag
+
+
+class EccCircuitRow:
+    """
+    ECC circuit
+    """
+
+    row: EccTableRow
+
+    ecc_chip: ECCVerifyChip
+
+    def __init__(self, row: EccTableRow, ecc_chip: ECCVerifyChip) -> None:
+        self.row = row
+        self.ecc_chip = ecc_chip
+
+    @classmethod
+    def check_fq(cls, value: int) -> bool:
+        return value < int(FP.field_modulus)
+
+    @classmethod
+    def assign(
+        cls,
+        op_type: EccOpTag,
+        p: List[Tuple[Word, Word]],
+        out: Tuple[FP, FP],
+    ):
+        if op_type == EccOpTag.Add:
+            return cls.assign_add(p[0], p[1], out)
+        elif op_type == EccOpTag.Mul:
+            return cls.assign_mul(p[0], p[1], out)
+        elif op_type == EccOpTag.Pairing:
+            return cls.assign_pairing(p, out)
+        else:
+            raise TypeError(f"Not supported type: {op_type}")
+
+    @classmethod
+    def assign_add(cls, p0: Tuple[Word, Word], p1: Tuple[Word, Word], out: Tuple[FP, FP]):
+        # 1. verify validity of input points p0 and p1
+        precheck_p0x = cls.check_fq(p0[0].int_value())
+        precheck_p0y = cls.check_fq(p0[1].int_value())
+        precheck_p1x = cls.check_fq(p1[0].int_value())
+        precheck_p1y = cls.check_fq(p1[1].int_value())
+
+        point0 = (FP(p0[0].int_value()), FP(p0[1].int_value()))
+        point1 = (FP(p1[0].int_value()), FP(p1[1].int_value()))
+        is_valid_p0 = is_on_curve(point0, b)
+        is_valid_p1 = is_on_curve(point1, b)
+        is_infinite_p0 = is_inf(point0)
+        is_infinite_p1 = is_inf(point1)
```

Comment:
It seems `is_on_curve` already checks if `is_inf`, but the `is_inf` returns `True` only if the input is `None`, so I think here they would never be `True`.

Reply:
fixed in [b6c812d](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/500/commits/b6c812d260d279ec95bbd8f9525d58dcbce75559)